### PR TITLE
Including condition explanation into CollectionCondition toString()

### PR DESCRIPTION
**Pull request by rkliuha**
## Proposed changes
These changes add condition explanation into CollectionCondition toString() as it is in single Condition.
It's needed for better events logging because LogEvent.toString() for collections verification does not include explanation message (because)

## Checklist
- [x] Checkstyle and unit tests pass locally with my changes by running `gradle check chrome htmlunit` command
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/main/java/com/codeborne/selenide/CollectionCondition.java [hunk 1]
```diff
@@ -1,16 +1,9 @@
...
-import com.codeborne.selenide.collections.ExactTexts;
-import com.codeborne.selenide.collections.ListSize;
-import com.codeborne.selenide.collections.SizeGreaterThan;
-import com.codeborne.selenide.collections.SizeGreaterThanOrEqual;
-import com.codeborne.selenide.collections.SizeLessThan;
-import com.codeborne.selenide.collections.SizeLessThanOrEqual;
-import com.codeborne.selenide.collections.SizeNotEqual;
-import com.codeborne.selenide.collections.Texts;
-import com.codeborne.selenide.collections.TextsInAnyOrder;
+import com.codeborne.selenide.collections.*;
```
Comment (rosolko) [inline src/main/java/com/codeborne/selenide/CollectionCondition.java]: Why don't you run gradle `check` task locally?
This import changes spawn an error:
```
[ant:checkstyle] [ERROR] /home/travis/build/selenide/selenide/src/main/java/com/codeborne/selenide/CollectionCondition.java:3: Using the '.*' form of import should be avoided - com.codeborne.selenide.collections.*. [AvoidStarImport]
```

File: src/main/java/com/codeborne/selenide/CollectionCondition.java [hunk 2]
```diff
@@ -103,12 +96,42 @@ public static CollectionCondition exactTexts(List<String> expectedTexts) {
     return new ExactTexts(expectedTexts);
   }
 
+  private static class ExplainedCollectionCondition extends CollectionCondition {
```
Comment (rosolko) [inline src/main/java/com/codeborne/selenide/CollectionCondition.java]: Also new class and method lack of documentation.

**Comment by rosolko** [comment]
@rkliuha Please take a look at the pull request check list

File: src/main/java/com/codeborne/selenide/CollectionCondition.java (hunk 1 above)
Comment (rkliuha) [reply src/main/java/com/codeborne/selenide/CollectionCondition.java]: I've encountered with some checkstyle issue while running it locally. Now It's fixed. Thank you

File: src/main/java/com/codeborne/selenide/CollectionCondition.java (hunk 2 above)
Comment (rkliuha) [reply src/main/java/com/codeborne/selenide/CollectionCondition.java]: Added overall explanation

**Comment by rkliuha** [comment]
@rosolko Thank you for review! Tests on CI are still failing but I'm not sure that it's my changes

**Comment by symonk** [comment]
I don't think your changes are breaking tests.  Firefox on travis has been problematic for quite a while, we should fix it :) chrome is surprisingly failing on proxy/download/auth related issues

**Pull request merged**